zerver/tests/test_narrow.py

```diff
@@ -880,6 +880,10 @@ def test_get_messages_with_search(self):
             ('meetings', 'please bring your laptops to take notes'),
             ('dinner', 'Anybody staying late tonight?'),
             ('urltest', 'https://google.com'),
+            (u'日本', u'こんにちは。今日はいい天気ですね。'),
+            (u'日本', u'今朝はごはんを食べました。'),
+            (u'日本', u'昨日、日本 のお菓子を送りました。'),
```

Comment:
I think we want at least one test that has multiple matches, since we have fiddly arithmetic with offsets in our loop.